### DISCUSSION TITLE
Userscript engine on bromite doesn't work well

### DISCUSSION THREAD

**Discussion by ziadhussain007** [bromite/bromite#2300] [Builds]
I'm using bromite browser v 104 latest , The problem of youtube ads still exist although enabling adblock and cosmetic filter userscipt , so I tried an adguard youtube adblock userscript I found on greasyfork but it didn't work , the same userscript I tried it on kiwi brower and via browser ( both browsers have userscript support) it worked fine on the both browser but it didn't work in bromite

**Comment by uazo** [comment]
do you mean https://github.com/ChenZihan-sudo/Youtube-AD-blocker/blob/main/new%20feature/0.6.js ?
with bromite the error is
```
Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the 
following Content Security Policy directive: "script-src 'self'".
```
it is a precise choice of bromite of not allowing `eval()` in userscripts

**Reply by ziadhussain007** [reply]
No bro I didn't mean this , I meant that script
https://greasyfork.org/en/scripts/436535-adguard-script-block-youtube-ads

**Reply by uazo** [reply]
it's the same:

`Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'self'". Either the 'unsafe-inline' keyword, a hash ('sha256-='), or a nonce ('nonce-...') is required to enable inline execution.`

although it seems simple to fix. if you edit
```
    const script = document.createElement('script');
    const scriptText = pageScript.toString().replace('__logo_text__', getMessage('logo'));
    script.innerHTML = `(${scriptText})();`;
    document.head.appendChild(script);
    document.head.removeChild(script);
```
to simple
```
pageScript();
```
(so remove the eval) it seems to work

**Reply by ziadhussain007** [reply]
Can you edit it for me bro and upload it

**Reply by ziadhussain007** [reply]
I modified the script as you said bro and the ads didn't appear , but instead of that a black screen is showing for 5 seconds during the ad period before the main video appear can you solve this too

**Reply by ziadhussain007** [reply]
Can you adjust a script especially for youtube ads??
